torch/distributions/kl.py

```diff
@@ -278,6 +307,21 @@ def _kl_laplace_laplace(p, q):
     return t1 + t2 + t3 - 1
 
 
+@register_kl(MultivariateNormal, MultivariateNormal)
+def _kl_multivariatenormal_multivariatenormal(p, q):
+    # From https://en.wikipedia.org/wiki/Multivariate_normal_distribution#Kullback%E2%80%93Leibler_divergence
+    if p.event_shape != q.event_shape:
+        raise ValueError("KL-divergence between two Multivariate Normals with\
+                          different event shapes cannot be computed")
+
+    term1 = _batch_diag(q.scale_tril).log().sum(-1) - _batch_diag(p.scale_tril).log().sum(-1)
+
+    term2 = _batch_trace(_batch_mm(_batch_inverse(q.covariance_matrix), p.covariance_matrix))
```

Comment:
I am using this is instead of `q.precision_matrix`. Something seems to have been broken between current HEAD and https://github.com/pytorch/pytorch/commit/c2703aa1412fa07f2dc909abda635770a5c62792 .

Reply:
@fehiepsi do you think your #6128 may be related?

Reply:
In my opinion, it is better to use `q.scale_tril` to calculate things related to the inverse of a positive definite matrix.

In this case, we can simplify the computation by using properties of trace:
$ tr( inv(LLt) @ SSt) = tr( inv(Lt) inv(L) S St) = tr( inv(L) S St inv(Lt)) $
So this trace term can be calculated from inv(L) S by squaring it and take sum.